Add development docs for Security Analytics (#8463)
wazuh/wazuh-kibana-app: docs/dev/modules/security-analytics.md

# Security Analytics

> This module allows to manage the analysis of the security events logs and generate alerts based on the defined rules. It provides a set of tools to create, edit, and manage the decoders, integrations, filters, KVDBs, detectors and rules and test the logs against the defined integrations in logtest.

Reviewer: Small typos.

```suggestion
This module allows us to manage the security events logs analysis, and generate alerts based on the defined rules. It provides a set of tools to create, edit, and manage decoders, integrations, filters, KVDBs, detectors and rules, and test the logs against the defined integrations in logtest.
```

Reviewer 2: I’d prefer not to apply the suggestion about "us" because our docs use an impersonal/third-person tone (for example, “This module…”, “This app…”), not first person “us”.

Reviewer 2: resolved [here](https://github.com/wazuh/wazuh-dashboard-plugins/pull/8463/commits/f5bd32080b4ef4420e7ca7a56328a2c0898d64e1)

> The feature is provided by the [wazuh-dashboard-security-analytics](https://github.com/wazuh/wazuh-dashboard-security-analytics) plugin, that is based in the OpenSearch Dashboards plugin and adapted to our requirements. This is included in the distribution of the Wazuh dashboard.

Reviewer: Small typo.

```suggestion
The feature is provided by the [wazuh-dashboard-security-analytics](https://github.com/wazuh/wazuh-dashboard-security-analytics) plugin based on the OpenSearch Dashboards plugin and adapted to our requirements. This is included in the distribution of the Wazuh dashboard.
```

Reviewer 2: resolved [here](https://github.com/wazuh/wazuh-dashboard-plugins/pull/8463/commits/f5bd32080b4ef4420e7ca7a56328a2c0898d64e1)

- **Cloud Services**: Integrations related to the analysis of cloud services logs, such as services managed by cloud providers.
- **Network Activity**: Integrations related to the analysis of network activity logs, such as DNS, HTTP, Email, SSH, FTP, DHCP, RPD.
- **System Activity**: Integrations related to the analysis of system logs, such as system monitoring logs.
- **Security**: Integrations related to the analysis of security logs, such as security-related logs and events.
- **Others**: Integrations that do not fit in the previous categories.

## Filter

A filter is a set of conditions that allows to exclude events based on specific criteria of the analysis flow.

Store: `wazuh-threatintel-filters` index.

Filter has the following properties:
| Property | Description |
| --------------- | -------------------------------------------------------- |
| `name` | The name of the filter. |
| `description` | A description of the filter. |
| `type` | The type of the filter. It can be `pre-filter` or `post-filter`. |
| `enabled` | Indicates if the filter is enabled or not. |

> | `check expression` | The check expression that defines the conditions of the filter. |

Reviewer: Small change to align with changes in: [Filters PR](github.com/wazuh/wazuh-dashboard-security-analytics/pull/293)

```suggestion
| `check` | An array of objects, e.g., [{field: value}, ...], that defines the filter conditions.
```

Reviewer 2: resolved [here](https://github.com/wazuh/wazuh-dashboard-plugins/pull/8463/commits/f5bd32080b4ef4420e7ca7a56328a2c0898d64e1)
